src/components/Shared/Markdown/index.tsx

```diff
@@ -0,0 +1,85 @@
+import LexicalAutoLinkPlugin, { URL_MATCHER } from '@components/Shared/Lexical/Plugins/AutoLinkPlugin';
+import type { LensterPublication } from '@generated/lenstertypes';
+import { $createLinkNode, LinkNode } from '@lexical/link';
+import { $convertFromMarkdownString, TEXT_FORMAT_TRANSFORMERS } from '@lexical/markdown';
+import { useLexicalComposerContext } from '@lexical/react/LexicalComposerContext';
+import { ContentEditable } from '@lexical/react/LexicalContentEditable';
+import { HashtagPlugin } from '@lexical/react/LexicalHashtagPlugin';
+import { HistoryPlugin } from '@lexical/react/LexicalHistoryPlugin';
+import { RichTextPlugin } from '@lexical/react/LexicalRichTextPlugin';
+import type { ElementNode, LexicalNode } from 'lexical';
+import { $createTextNode } from 'lexical';
+import type { FC } from 'react';
+import { ERROR_MESSAGE } from 'src/constants';
+
+import MentionsPlugin, { AtSignMentionsRegex } from '../../Shared/Lexical/Plugins/AtMentionsPlugin';
+import { $createMentionNode, MentionNode } from '../Lexical/Nodes/MentionsNode';
+
+const TRANSFORMERS = [
+  ...TEXT_FORMAT_TRANSFORMERS,
+  {
+    dependencies: [MentionNode],
+    export: () => {
+      return null;
+    },
+    regExp: AtSignMentionsRegex,
+    replace: (parentNode: ElementNode, children: Array<LexicalNode>, match: Array<String>) => {
+      // we have to check if the match is a valid mention and then convert it to mentionNode
+      const node = $createMentionNode(match[0].split('@')[1]);
```

Comment:
@bigint we have to check if all mentions are a valid mention and then convert it to mention node. Add proper logic here?

Reply:
Hmm I think we can better catch mentions by regex 🤔 

If we use this it will break in many places, for example TikTok URL will look something like this https://tt.com/@strek/493849385983

Reply:
No No test the regex with the link. It wont detect this as mention. 

<img width="849" alt="Screenshot 2022-11-14 at 11 54 22 AM" src="https://user-images.githubusercontent.com/32865581/201589674-093db66c-a82a-4fac-8576-932581ec76d8.png">

Reply:
cool then 🚀